external/llvm-project/mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -326,6 +326,62 @@ LogicalResult MFMAOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DPPOp
+//===----------------------------------------------------------------------===//
+LogicalResult DPPOp::verify() {
+  Type srcType = getSrc().getType();
+  if (srcType.getIntOrFloatBitWidth() > 32) {
+    return emitOpError("integer and floating point types larger than 32 bits "
+                       "are not supported");
+  }
+
+  DPPPerm kind = getKind();
+  auto permArgument = getPermArgument();
```

Comment:
I'd explicitly write `std::optional<Attribute>` here if that's the return type.

Heck, I'd go for `Attribute permArgument = getPermArgument().value_or(Attribute{});`